reflex-dhtmlx/src/Reflex/Dom/DHTMLX/Date.hs

```diff
@@ -100,10 +102,12 @@ dhtmlxDatePicker
 dhtmlxDatePicker (DatePickerConfig iv sv b p wstart attrs visibleOnLoad) = do
     let fmt = "%Y-%m-%d"
         formatter = T.pack . maybe "" (formatTime defaultTimeLocale fmt)
+        ivTxt = formatter iv
+    pb <- getPostBuild
     ti <- textInput $ def
       & attributes .~ attrs
-      & textInputConfig_initialValue .~ formatter iv
-      & textInputConfig_setValue .~ fmap formatter sv
+      & textInputConfig_initialValue .~ ivTxt
+      & textInputConfig_setValue .~ leftmost [formatter <$> sv, ivTxt <$ pb]
```

Comment:
Why would you need to set the value on post-build to exactly the initial value?

Reply:
So the dhtmlx calendar has a default time format, and it doesn't include time. We also cannot instantiate the dhtmlx-calendar with that format, it has to be set via a method after the object is `new`ed up. This results in a race condition. The calendar reads its value from the input field. The result is that dhtmlx parses the string without the time, resulting in `00:00`. Because postbuild is async, this allows the input to get kicked on the next tick, where the format is correctly set, and it will parse correctly.